Issues with Svelte 4

### Version:

- `@inertiajs/svelte` version: 1.0.9
- `svelte` version: 4

### Describe the problem:

When using Svelte 4, props are not being populated on inertia router visits. Also when adding svelte-eslint to the project pages are completely empty after visit.

### Steps to reproduce:

Just make a visit using inertia link or router to the route that requires any props

I am getting errors (not using TS).

```
    ~/Pr/inertia-bug-report    main !4  npm run dev                                                         1 ✘  7s  

> dev
> vite


  VITE v5.0.12  ready in 217 ms

  ➜  Local:   http://localhost:5173/
  ➜  Network: use --host to expose
  ➜  press h + enter to show help
✘ [ERROR] /Users/harry/Projects/inertia-bug-report/node_modules/@westacks/inertia-svelte/dist/components/Link.svelte:2:14 Unexpected token [plugin vite-plugin-svelte:optimize-svelte]

    node_modules/@westacks/inertia-svelte/dist/components/Link.svelte:2:14:
      2 │  import type { Method, PreserveStateOption, RequestPayload } from '@inertiajs/core'
        ╵               ^

  The plugin "vite-plugin-svelte:optimize-svelte" was triggered by this import

    node_modules/@westacks/inertia-svelte/dist/index.js:1:67:
      1 │ ..."@inertiajs/core";import{default as Te}from"./components/Link.svelte";import{router as v,setupProgress as P}from...
        ╵                                               ~~~~~~~~~~~~~~~~~~~~~~~~~~

✘ [ERROR] /Users/harry/Projects/inertia-bug-report/node_modules/@westacks/inertia-svelte/dist/components/App.svelte:2:14 Unexpected token [plugin vite-plugin-svelte:optimize-svelte]

    node_modules/@westacks/inertia-svelte/dist/components/App.svelte:2:14:
      2 │  import type { Page } from '@inertiajs/core'
        ╵               ^

  The plugin "vite-plugin-svelte:optimize-svelte" was triggered by this import

    node_modules/@westacks/inertia-svelte/dist/index.js:1:167:
      1 │ ...ess as P}from"@inertiajs/core";import E from"./components/App.svelte";import k from"./components/SSR.svelte";imp...
        ╵                                                ~~~~~~~~~~~~~~~~~~~~~~~~~

✘ [ERROR] /Users/harry/Projects/inertia-bug-report/node_modules/@westacks/inertia-svelte/dist/components/SSR.svelte:2:14 Unexpected token [plugin vite-plugin-svelte:optimize-svelte]

    node_modules/@westacks/inertia-svelte/dist/components/SSR.svelte:2:14:
      2 │  import type { Page } from '@inertiajs/core'
        ╵               ^

  The plugin "vite-plugin-svelte:optimize-svelte" was triggered by this import

    node_modules/@westacks/inertia-svelte/dist/index.js:1:206:
      1 │ ... from"./components/App.svelte";import k from"./components/SSR.svelte";import F from"./store";async function S({i...
        ╵                                                ~~~~~~~~~~~~~~~~~~~~~~~~~


  LARAVEL v10.43.0  plugin v1.0.1

  ➜  APP_URL: http://localhost
```

--- a/packages/svelte/src/Render.svelte
+++ b/packages/svelte/src/Render.svelte
@@ -14,12 +14,23 @@
   export let component
   export let props = {}
   export let children = []
+
+  let prevComponent;
+  let key;
+  $: {
+    if (prevComponent !== component) {
+      key = Date.now();
+      prevComponent = component;
+    }
+  }
 </script>
 
 {#if $store.component}
-  <svelte:component this={component} {...props}>
-    {#each children as child, index (component && component.length === index ? $store.key : null)}
-      <svelte:self {...child} />
-    {/each}
-  </svelte:component>
+  {#key key}
+    <svelte:component this={component} {...props}>
+      {#each children as child, index (component && component.length === index ? $store.key : null)}
+        <svelte:self {...child} />
+      {/each}
+    </svelte:component>
+  {/key}
 {/if}

--- a/packages/svelte/src/Render.svelte
+++ b/packages/svelte/src/Render.svelte
@@ -15,12 +15,12 @@
   export let props = {}
   export let children = []
 
-  let prevComponent;
-  let key;
+  let prevComponent
+  let key
   $: {
     if (prevComponent !== component) {
-      key = Date.now();
-      prevComponent = component;
+      key = Date.now()
+      prevComponent = component
     }
   }
 </script>

